tests/wasm_tests/tic_tac_toe_tests.cpp

```diff
@@ -0,0 +1,211 @@
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-compare"
+#include <boost/test/unit_test.hpp>
+#pragma GCC diagnostic pop
+#include <boost/algorithm/string/predicate.hpp>
+#include <eosio/testing/tester.hpp>
+#include <eosio/chain/contracts/abi_serializer.hpp>
+
+#include <tic_tac_toe/tic_tac_toe.wast.hpp>
+#include <tic_tac_toe/tic_tac_toe.abi.hpp>
+#include <eosio.system/eosio.system.wast.hpp>
+#include <eosio.system/eosio.system.abi.hpp>
+
+#include <proxy/proxy.wast.hpp>
+#include <proxy/proxy.abi.hpp>
+
+#include <Runtime/Runtime.h>
+
+#include <fc/variant_object.hpp>
+#include <fc/io/json.hpp>
+
+#ifdef NON_VALIDATING_TEST
+#define TESTER tester
+#else
+#define TESTER validating_tester
+#endif
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::chain::contracts;
+using namespace eosio::testing;
+using namespace fc;
+
+struct movement {
+   uint32_t    row;
+   uint32_t    column;
+
+   template<typename DataStream>
+   friend DataStream& operator << ( DataStream& ds, const movement& t ){ \
+      return ds << t.row << t.column;
+   }
+   template<typename DataStream>
+   friend DataStream& operator >> ( DataStream& ds, movement& t ){
+      return ds >> t.row >> t.column;
+   }
+};
+
+BOOST_AUTO_TEST_SUITE(tic_tac_toe_tests)
+
+BOOST_AUTO_TEST_CASE( tic_tac_toe_game ) try {
+   TESTER chain;
+   abi_serializer abi_ser = json::from_string(tic_tac_toe_abi).as<abi_def>();
+
+   chain.set_code(config::system_account_name, eosio_system_wast);
+   chain.set_abi(config::system_account_name, eosio_system_abi);
```

Comment:
Do you need to set the system account for this to work?

Reply:
No, removed